dist_tables/ddl.rst

```diff
@@ -61,6 +61,45 @@ Each created shard is assigned a unique shard id and all its replicas have the s
 
 You are now ready to insert data into the distributed table and run queries on it. You can also learn more about the UDF used in this section in the :ref:`user_defined_functions` of our documentation.
 
+.. _colocation_groups:
+
+Co-Location Groups
+------------------
+
+By default two tables are co-located when they are distributed by columns of the same type using the same distribution method and over the same number of shards. Another way to say this is that the tables are in the same *co-location group.*
+
+It is important for performance to have full control over co-location groups. All members of a co-location group support router-executed JOIN queries between one another on their distribution column. They also support foreign key constraints for keys containing the distribution column. Additionally shard rebalancing preserves co-location in new shard placements.
+
+To control co-location groups manually use the optional :code:`colocate_with` parameter of :code:`create_distributed_table`. Left unspecified it defaults to the value :code:`default` which uses the behavior described above to lump all tables having the same distribution column type into the same co-location group. These statements are equivalent:
+
+.. code-block:: sql
+
+  SELECT create_distributed_table('github_events', 'repo_id');
+
+  --
+  -- is equivalent to
+  --
+
+  SELECT create_distributed_table('github_events', 'repo_id', colocate_with => 'default');
+
+
+To start a new group and add tables to it use the two other modes of :code:`colocate_with`: the reserved string :code:`none` and the name of another table.
+
+.. code-block:: sql
+
+  -- start a new group
+  SELECT create_distributed_table('products', 'store_id', colocate_with => 'none');
+
+  -- add to the same group as products
+  SELECT create_distributed_table('orders', 'store_id', colocate_with => 'products');
+
+There are a few reasons to create new co-location groups:
+
+* To express that tables distributed by same type of columns using the same shard count should *not* be related. This makes the rebalancer more efficient because it has more flexibility in placing shards.
```

Comment:
By "expressing modeling intentions" I had in mind the DDL commands as source code. When reading the code it is more apparent that the author intended two tables to be colocated when they say so with colocate_with. My premise there might be incorrect though, that people share and refer to the DDL commands for a schema. In reality maybe somebody executes the DDL commands and afterwards other people, ignorant of the commands, interact with the database as it was created.